azurerm/resource_arm_application_gateway.go

```diff
@@ -1865,6 +1971,14 @@ func flattenApplicationGatewayRequestRoutingRules(input *[]network.ApplicationGa
 	for _, config := range *input {
 		if props := config.ApplicationGatewayRequestRoutingRulePropertiesFormat; props != nil {
 
+			if props.BackendAddressPool != nil && props.RedirectConfiguration != nil {
+				return nil, fmt.Errorf("[ERROR] Conflict between `backend_address_pool_name` and `redirect_configuration_name` (back-end pool not applicable when redirection specified)")
+			}
+
+			if props.BackendHTTPSettings != nil && props.RedirectConfiguration != nil {
```

Comment:
(same here)

Reply:
again, these are SubResource * - Feels like we need a predicate func like `ValidSubReference()` which returns true if pointing to a valid resource ID.

Reply:
we probably could split this out into a private function - but be aware in Golang private functions (starting with a lower-case character) are shared throughout the package - so the naming would need to be more `applicationGatewayValidateSubResource(...)`

Reply:
done